noggin/tests/unit/controller/test_authentication.py

```diff
@@ -109,6 +109,44 @@ def test_login_no_username(client):
     assert "noggin_username" not in session
 
 
+def test_login_username_case(client, dummy_user):
+    """Test giving username random uppercase letter"""
+    result = client.post(
+        '/',
+        data={
+            "login-username": "duMmy",
+            "login-password": "dummy_password",
+            "login-submit": "1",
+        },
+        follow_redirects=True,
+    )
+    page = BeautifulSoup(result.data, 'html.parser')
+    messages = page.select(".flash-messages .alert-success")
+    assert len(messages) == 1
+    assert messages[0].get_text(strip=True) == 'Welcome, dummy!×'
+    assert session.get("noggin_username") == "dummy"
+    assert session.get("noggin_session") is not None
+    
+
+def test_login_username_created_with_case(client, dummy_user_with_case):
+    """Test giving username random uppercase letter"""
```

Comment:
This should be changed to reflect the test, looks like it was copied from the other test.